New entity id/uuid procedure blocks

AgnyaAlm opened the pull request:
This PR adds 2 new procedure blocks.

- Get (entity) 's UUID (Forge 1.16.5 and 1.18.2)
- Get (entity) 's namespaced id (Forge 1.16.5 and 1.18.2)

These blocks are taken from my plugin for 2022.1 ([Entity ID & UUID Procedures](https://mcreator.net/plugin/89039/namespace-id-uuid-procedures)

- [x] Add procedure blocks
- [x] Test if they work properly
- [x] Add translation strings to text.properties
- [x] Fix test fail when adding the translation strings

Goldorion commented on plugins/generator-1.18.2/forge-1.18.2/procedures/entity-freeze.java.ftl:
```diff
@@ -0,0 +1,2 @@
+
```
This line is unnecessary.

AgnyaAlm replied:
I added it because on my tests the code was put in the same line if that line wasn't added. Tho it works without it, it still might confuse users imo

Goldorion commented on plugins/mcreator-core/procedures/entity-freeze.json:
```diff
@@ -0,0 +1,30 @@
+{
+   "message0": "Freeze %2 for %1 ticks",
```
Put messages inside text files, so they can be translated.

AgnyaAlm replied:
Sure thing, gonna work on that!

AgnyaAlm replied:
Could you give me an example on where to implement that?

Goldorion replied:
https://github.com/MCreator/MCreator/blob/master/plugins/mcreator-localization/lang/texts.properties#L762

MagicalAlexey commented:
you're good, that's just what I needed)

AgnyaAlm commented:
Renamed blocks to fit the rest (ex. from get_entity_uuid to entity_get_uuid) and added all blocks to the text.properties file. I'm assuming I don't have to add them to the rest of the files as Crowdin should take care of them iirc, but I might be wrong.

KlemenDEV commented:
1. Tests are failing
2. Please split falling procedures into another PR as I would wait until 1.16.5 is dropped to merge them

KlemenDEV commented:
> /tmp/mcreator_test_workspace3540908924055557950/src/main/java/net/mcreator/test_mod/procedures/TestProcedureBlockentity_freezeProcedure.java:17: error: incompatible types: possible lossy conversion from double to int
    		entity.setTicksFrozen(test);
    		                      ^

You need to do proper type casting. Check test logs to see the problem

KlemenDEV commented:
>I've tested it with the last commit using the runMCreator gradle task and it worked with no issues yet still it's failing. What is the issue?

This is not unit test, you need to run unit tests to test all the functionality too

AgnyaAlm commented:
Forcing ticking into an int worked.

KlemenDEV reviewed (changes requested):
For casting to int, use toInt system we use in other procedures.

Also I suggest you keep 1.18.2 features for later PR, or keep this PR draft until 1.16.5 is dropped (which may take a while, so better to split the PR)

AlsoSomeoneElse commented on plugins/mcreator-core/procedures/entity_get_id.json:
```diff
@@ -0,0 +1,23 @@
+{
+"args0": [
```
Please change the indentation so that it matches other procedure .json files

AlsoSomeoneElse commented on plugins/mcreator-localization/lang/texts.properties:
```diff
@@ -877,7 +877,9 @@ blockly.block.entity_get_absorption=Get absorption health level of %1
 blockly.block.entity_get_armor_slot_item=Get item from armor slot %1 of %2
 blockly.block.entity_get_controlling_passenger=%2 Get passenger that is controlling %1
 blockly.block.entity_get_fly_speed=Get fly speed of %1
+blockly.block.entity_get_id=Get %1 Namespaced ID
```
I'd change the texts to "Get namespaced ID of %1" and "Get UUID of %1"

navarayuk replied:
isn't the namespace ID the mod:item so minecraft:pig ?

Defeatomizer replied:
He meant _"Get %1 Namespaced ID" to "Get namespaced ID of %1" and "Get %1 Universal Unique Identifier (UUID)" to "Get UUID of %1"_.
